packages/react-router/src/route.ts

```diff
@@ -220,14 +227,14 @@ export type UpdatableRouteOptions<
   onLeave?: (match: AnyRouteMatch) => void
   meta?: (ctx: {
     params: TAllParams
-    loaderData: TLoaderData
+    loaderData: TLoaderData extends never ? undefined : TLoaderData
```

Comment:
when would `TLoaderData` be `never`?
Shouldn't this be already caught by the `TLoaderDataReturn` pattern?